Applet/JCardSimProvider/test/com/android/javacard/test/KMRKPFunctionalTest.java

```diff
@@ -405,17 +405,36 @@ public void testGenerateCsr(short no_keys) {
     payloadLen = KMKeymasterApplet.encodeToApduBuffer(signStructure, payload,
         (short) 0, KMKeymasterApplet.MAX_COSE_BUF_SIZE);
 
-    short payloadByteLen = (short) ((short) 1 /*Array of 5 elements occupies 1 byte */ +
+    short csrPayloadLen = (short)(1/*Array of 4 elements occupies 1 byte*/
+        + 1 /*version*/ + encoder.getEncodedBytesLength(certTypeLen) + certTypeLen + deviceInfoBytesLen
+        + coseKeyBytesLen);
+    short csrPayloadByteLen = encoder.getEncodedBytesLength(csrPayloadLen);
```

Comment:
Oh, I saw it here. Ideally this should be with the rkp specification changes pull request right ?